workflow-ui/compose/src/androidTest/java/com/squareup/workflow1/ui/compose/ComposeViewTreeIntegrationTest.kt

```diff
@@ -588,24 +589,31 @@ internal class ComposeViewTreeIntegrationTest {
       context: Context,
       container: ViewGroup?
     ): ScreenViewHolder<TestComposeRendering> {
-      val view = ComposeView(context)
-      return ScreenViewHolder(initialEnvironment, view) { rendering, _ ->
-        val lastCompositionStrategy = view.tag as? ViewCompositionStrategy
-        view.tag = rendering.disposeStrategy
-        if (rendering.disposeStrategy != lastCompositionStrategy) {
-          lastCompositionStrategy?.let { view.setViewCompositionStrategy(it) }
-        }
+      var lastCompositionStrategy = initialRendering.disposeStrategy
+
+      return ComposeView(context).let { view ->
+        lastCompositionStrategy?.let(view::setViewCompositionStrategy)
+
+        ScreenViewHolder(initialEnvironment, view) { rendering, _ ->
+          if (rendering.disposeStrategy != lastCompositionStrategy) {
+            lastCompositionStrategy = rendering.disposeStrategy
+            lastCompositionStrategy?.let { view.setViewCompositionStrategy(it) }
```

Comment:
Oh! I just started looking into `setViewCompositionStrategy` with some test leaks we're getting. I'm gonna be looking into if we're setting this in prod and why